notify/impl.go

```diff
@@ -268,34 +363,25 @@ func (n *PagerDuty) Notify(ctx context.Context, as ...*types.Alert) error {
 
 	log.With("incident", key).With("eventType", eventType).Debugln("notifying PagerDuty")
 
-	groupLabels, ok := GroupLabels(ctx)
-	if !ok {
-		log.Error("missing group labels")
-	}
-
-	data := struct {
-		Alerts      model.Alerts
-		GroupLabels model.LabelSet
-	}{
-		Alerts:      alerts,
-		GroupLabels: groupLabels,
-	}
-
 	var err error
 	tmpl := func(name string) (s string) {
 		if err != nil {
 			return
 		}
-		s, err = n.tmpl.ExecuteTextString(name, &data)
+		s, err = n.tmpl.ExecuteTextString(name, data)
 		return s
 	}
+	details := make(map[string]string, len(n.conf.Details))
+	for k, v := range n.conf.Details {
+		details[k] = tmpl(v)
+	}
 
 	msg := &pagerDutyMessage{
-		ServiceKey:  n.conf.ServiceKey,
+		ServiceKey:  tmpl(n.conf.ServiceKey),
```

Comment:
I can imagine routing being plumbed through from service discovery. I had a system many years back that used this technique very successfully for email, and can see the same being done for pagerduty etc.